spec/services/provider_interface/filter_application_choices_for_providers_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'rails_helper'
+
+RSpec.describe FilterApplicationChoicesForProviders do
+  include CourseOptionHelpers
+
+  describe '#call' do
+    let(:submitted_application_choice) { create(:application_choice, :awaiting_provider_decision, status: 'awaiting_provider_decision') }
+    let(:offered_application_choice) { create(:application_choice, :with_offer, status: 'offer') }
+    let(:accepted_application_choice) { create(:application_choice, :with_accepted_offer, status: 'pending_conditions') }
+    let(:conditions_met_application_choice) { create(:application_choice, :awaiting_provider_decision, status: 'recruited') }
+    let(:enrolled_application_choice) { create(:application_choice, status: 'enrolled') }
+    let(:rejected_application_choice) { create(:application_choice, :with_rejection, status: 'rejected') }
+    let(:declined_application_choice) { create(:application_choice, :with_declined_offer, status: 'declined') }
+    let(:application_withdrawn_application_choice) { create(:application_choice, status: 'withdrawn') }
+    let(:conditions_not_met_application_choice) { create(:application_choice, status: 'conditions_not_met') }
+    let(:withdrawn_by_us_application_choice) { create(:application_choice, status: 'rejected', offer_withdrawn_at: 2.days.ago) }
+
+    context 'when filtering by status' do
+      let(:application_choices) { ApplicationChoice.all }
+
+      test_data =
+        [
+          { filter: 'Submitted', filter_status: 'awaiting_provider_decision' },
+          { filter: 'Offered', filter_status: 'offer' },
+          { filter: 'Accepted', filter_status: 'pending_conditions' },
+          { filter: 'Conditions met', filter_status: 'recruited' },
+          { filter: 'Enrolled', filter_status: 'enrolled' },
+          { filter: 'Rejected', filter_status: 'rejected' },
+          { filter: 'Declined', filter_status: 'declined' },
+          { filter: 'Application withdrawn', filter_status: 'withdrawn' },
+          { filter: 'Conditions not met', filter_status: 'conditions_not_met' },
+          { filter: 'Withdrawn by us', filter_status: 'offer_withdrawn' },
+        ]
+
+      test_data.each_with_index do |example, index|
+        it "single #{example[:filter]} filter returns only #{example[:filter].downcase} application choices" do
+          subject = described_class.call(
```

Comment:
`subject` has special meaning in RSpec and there's usually a better name for whatever it is — `returned_application_choices` or sth?